src/test/v1.2/advanced/channels-support-1.2.ts

```diff
@@ -121,23 +125,27 @@ export class ChannelControl1_2 implements ChannelControl<Channel, Context> {
 
   setupAndValidateListener2 = (channel: Channel, expectedContextType: string, errorMessage: string, onComplete: (ctx: Context) => void): void => {
     if (channel) {
-      listener2 = channel.addContextListener(null, (context) => {
-        expect(context.type).to.be.equals(expectedContextType, errorMessage);
+      listener2 = channel.addContextListener(expectedContextType, (context) => {
+        if (expectedContextType != null) {
+          expect(context.type).to.be.equals(expectedContextType, errorMessage);
+        }
         onComplete(context);
       });
     } else {
-      listener2 = fdc3.addContextListener(null, (context) => {
-        expect(context.type).to.be.equals(expectedContextType, errorMessage);
+      listener2 = fdc3.addContextListener(expectedContextType, (context) => {
+        if (expectedContextType != null) {
+          expect(context.type).to.be.equals(expectedContextType, errorMessage);
+        }
         onComplete(context);
       });
     }
 
     validateListenerObject(listener2);
   }
 
-  setupContextChecker = async (channel: Channel, expectedContextType: string, errorMessage: string, onComplete: (ctx: Context) => void): Promise<void> => {
+  setupContextChecker = async (channel: Channel,  requestedContextType: string, expectedContextType: string, errorMessage: string, onComplete: (ctx: Context) => void): Promise<void> => {
     //Retrieve current context from channel
-    const context = await channel.getCurrentContext();
+    const context = (requestedContextType == undefined) ? await channel.getCurrentContext() : await channel.getCurrentContext(requestedContextType);
```

Comment:
```suggestion
    const context = (requestedContextType == null) ? await channel.getCurrentContext() : await channel.getCurrentContext(requestedContextType);
```
`requestedContextType ` is in teh middle of the args list so it will need to be `null` rather than `undefined`.